### PR TITLE
Update solarlog documentation based on new quality scale

### PR DESCRIPTION
**Pull request by dontinelli**
Update documentation to include new minimal requirements from quality scale.

## Proposed change

Update documentation to include new minimal requirements from quality scale (https://developers.home-assistant.io/docs/core/integration-quality-scale/).

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
	- Enhanced clarity and detail in the Solar-Log integration documentation.
	- Added sections for "Supported devices," "Configuration options," "Data updates," "Actions," and "Known limitations."
	- Emphasized the importance of password protection for the open JSON interface.
	- Clarified that the integration has no additional configuration options beyond the basic parameters.
	- Renamed "Sensors" to "Supported functionality" and corrected typographical errors.

### CODE REVIEW

File: source/_integrations/solarlog.markdown [hunk 1]
```diff
@@ -34,25 +38,41 @@ Please note that the open JSON interface only exposes a limited amount of data.
...
 
-## Additional template sensor
+To setup the integration you need the following information:
 
-In case you would like to get additional calculated sensors such as the amount of excess solar power available, you can use the [template platform](/integrations/template/).
+{% configuration_basic %}
+Name:
+    description: "The name for your Solar-Log device in Home Assistant. This will also be uses as prefix for the entity names."
+    required: true
+    type: string
```
Comment (frenck) [inline source/_integrations/solarlog.markdown]: `configuration_basic` does not have support for these parameters.

```suggestion
```

File: source/_integrations/solarlog.markdown [hunk 2]
```diff
@@ -34,25 +38,41 @@ Please note that the open JSON interface only exposes a limited amount of data.
...
+Host:
+    description: "The URL or IP address of your Solar-Log."
+    required: true
+    type: string
```
Comment (frenck) [inline source/_integrations/solarlog.markdown]:
```suggestion
```

File: source/_integrations/solarlog.markdown [hunk 3]
```diff
@@ -34,25 +38,41 @@ Please note that the open JSON interface only exposes a limited amount of data.
...
+User password available:
+    description: "Check the box, if you have the password for the Solar-Log user to securely access [all data](#additional-data)."
+    required: false
+    type: bool
```
Comment (frenck) [inline source/_integrations/solarlog.markdown]:
```suggestion
```

File: source/_integrations/solarlog.markdown [hunk 4]
```diff
@@ -34,25 +38,41 @@ Please note that the open JSON interface only exposes a limited amount of data.
...
+Password:
+    description: "Password for the Solar-Log user. Will only be asked for in a second step, if in the first step the checkbox has been selected."
+    required: false
+    type: string
```
Comment (frenck) [inline source/_integrations/solarlog.markdown]:
```suggestion
```

File: source/_integrations/solarlog.markdown [hunk 5]
```diff
@@ -98,6 +122,30 @@ In addition, information from devices connected to the Solar-Log device becomes
...
+In case you would like to get additional calculated sensors such as the amount of excess solar power available, you can use the [template platform](/integrations/template/).
+
+{% raw %}
+
+```yaml
+# Example configuration.yaml entry for sensor template platform
+template:
+  - sensor:
+    - name: "Solarlog yield day"
+      state: "{{ (states('sensor.solarlog_yield_day') | float(default=0) * 1000) | round(0,default=0) }}"
```
Comment (frenck) [inline source/_integrations/solarlog.markdown]: I'm not sure why one would use this sensor? As in, this takes an existing sensor and changes scale? A user can just select a different unit if they want to?

File: source/_integrations/solarlog.markdown [hunk 6]
```diff
@@ -98,6 +122,30 @@ In addition, information from devices connected to the Solar-Log device becomes
...
+
+{% endraw %}
+
+## Data updates
+
+The integration fetches data from the device every minute.
+
+## Actions
+
+The integration provides no actions.
```
Comment (frenck) [inline source/_integrations/solarlog.markdown]:
```suggestion
The integration provides no additional actions.
```

File: source/_integrations/solarlog.markdown [hunk 7]
```diff
@@ -34,25 +38,41 @@ Please note that the open JSON interface only exposes a limited amount of data.
...
+## Configuration options
 
-```yaml
-# Example configuration.yaml entry for sensor template platform
-template:
-  - sensor:
-    - name: "Solarlog yield day"
-      state: "{{ (states('sensor.solarlog_yield_day') | float(default=0) * 1000) | round(0,default=0) }}"
-```
+The integration has no configuration options.
```
Comment (frenck) [inline source/_integrations/solarlog.markdown]:
```suggestion
The integration has no additional configuration options.
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/solarlog.markdown (hunk 1 above)
Comment (dontinelli) [reply source/_integrations/solarlog.markdown]: Took this from the [documentation](https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/docs-installation-parameters/), this should then be amended...

File: source/_integrations/solarlog.markdown (hunk 5 above)
Comment (dontinelli) [reply source/_integrations/solarlog.markdown]: You're right. This example is from the "old" documentation. Will try to come up with a better example.

**Review by frenck — APPROVED** [review]
Thanks, @dontinelli 👍 

../Frenck

**Pull request merged**